### DISCUSSION TITLE
composite_demo 中的 code interpreter 获得答案后没有运行juyter

### DISCUSSION THREAD

**Discussion by a1120235890** [zai-org/ChatGLM3#459] [Bad Case]
### System Info / 系統信息

![1700720843747](https://github.com/THUDM/ChatGLM3/assets/57754489/5645815f-1cf2-48ec-aacb-ca6f2b78e32e)
在部署了composite_demo 后发现 这个code interpreter 没有像例子一样输出爱心图片，这个我应该怎么解决。我用的是chatglm3-6b模型。 跑到 with markdown_placeholder: 就没有继续了

### Information / 问题信息

- [X] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

windows下用 composite_demo  部署

### Expected behavior / 期待表现

在部署了composite_demo 后发现 这个code interpreter 没有像例子一样输出爱心图片，这个我应该怎么解决。我用的是chatglm3-6b模型。 跑到 with markdown_placeholder: 就没有继续了

**Comment by zRzRzRzRzRzRzR** [comment]
后台什么情况，对应的jupyter库装了吗

**Comment by a1120235890** [comment]
![image](https://github.com/THUDM/ChatGLM3/assets/57754489/8dee6045-5cfb-46cd-a588-b214476278c1)
这是后台的情况 然后我打印了一下 markdown_placeholder
![image](https://github.com/THUDM/ChatGLM3/assets/57754489/a0042300-9232-4439-91d7-260c35e50e04)

**Comment by a1120235890** [comment]
pip 环境：altair                    5.1.2
asttokens                 2.4.1
attrs                     23.1.0
blinker                   1.7.0
cachetools                5.3.2
certifi                   2023.11.17
charset-normalizer        3.3.2
click                     8.1.7
comm                      0.2.0
contourpy                 1.2.0
cycler                    0.12.1
debugpy                   1.8.0
decorator                 5.1.1
exceptiongroup            1.2.0
executing                 2.0.1
filelock                  3.13.1
fonttools                 4.45.0
fsspec                    2023.10.0
gitdb                     4.0.11
GitPython                 3.1.40
huggingface-hub           0.19.4
idna                      3.4
importlib-metadata        6.8.0
ipykernel                 6.27.0
ipython                   8.17.2
jedi                      0.19.1
Jinja2                    3.1.2
jsonschema                4.20.0
jsonschema-specifications 2023.11.1
jupyter_client            8.6.0
jupyter_core              5.5.0
kiwisolver                1.4.5
markdown-it-py            3.0.0
MarkupSafe                2.1.3
matplotlib                3.8.2
matplotlib-inline         0.1.6
mdurl                     0.1.2
mpmath                    1.3.0
nest-asyncio              1.5.8
networkx                  3.2.1
numpy                     1.26.2
nvidia-cublas-cu12        12.1.3.1
nvidia-cuda-cupti-cu12    12.1.105
nvidia-cuda-nvrtc-cu12    12.1.105
nvidia-cuda-runtime-cu12  12.1.105
nvidia-cudnn-cu12         8.9.2.26
nvidia-cufft-cu12         11.0.2.54
nvidia-curand-cu12        10.3.2.106
nvidia-cusolver-cu12      11.4.5.107
nvidia-cusparse-cu12      12.1.0.106
nvidia-nccl-cu12          2.18.1
nvidia-nvjitlink-cu12     12.3.101
nvidia-nvtx-cu12          12.1.105
packaging                 23.2
pandas                    2.1.3
parso                     0.8.3
pexpect                   4.8.0
Pillow                    10.1.0
pip                       23.3
platformdirs              4.0.0
prompt-toolkit            3.0.41
protobuf                  4.25.1
psutil                    5.9.6
ptyprocess                0.7.0
pure-eval                 0.2.2
pyarrow                   14.0.1
pydeck                    0.8.1b0
Pygments                  2.17.2
pyparsing                 3.1.1
python-dateutil           2.8.2
pytz                      2023.3.post1
PyYAML                    6.0.1
pyzmq                     25.1.1
referencing               0.31.0
regex                     2023.10.3
requests                  2.31.0
rich                      13.7.0
rpds-py                   0.13.1
safetensors               0.4.0
sentencepiece             0.1.99
setuptools                68.0.0
six                       1.16.0
smmap                     5.0.1
stack-data                0.6.3
streamlit                 1.28.2
sympy                     1.12
tenacity                  8.2.3
tokenizers                0.15.0
toml                      0.10.2
toolz                     0.12.0
torch                     2.1.1
tornado                   6.3.3
tqdm                      4.66.1
traitlets                 5.13.0
transformers              4.35.2
triton                    2.1.0
typing_extensions         4.8.0
tzdata                    2023.3
tzlocal                   5.2
urllib3                   2.1.0
validators                0.22.0
watchdog                  3.0.0
wcwidth                   0.2.12
wheel                     0.41.2
zipp                      3.17.0

**Comment by zRzRzRzRzRzRzR** [comment]
查看一下和 #198 有没有关系，检查jupyter内核地址，另外，检查一下其他的工具是否正常

**Comment by zRzRzRzRzRzRzR** [comment]
相似的Issue已经移交到讨论区，可以到讨论区讨论一下，目前无法复现，该issue关闭

**Comment by longerking** [comment]
我猜测是你的创建的conda 环境名称和默认名称不一样，而且没有把IPYKERNEL设置为环境变量。
比如Readme中：https://github.com/THUDM/ChatGLM3/blob/33953b119e794f4af81ce5200343eeea8c72994c/composite_demo/README.md?plain=1#L35

创建的环境为chatglm3-demo
```
conda create -n chatglm3-demo python=3.10
conda activate chatglm3-demo
pip install -r requirements.txt

ipython kernel install --name chatglm3-demo --user

# 确保 export IPYKERNEL=<kernel_name>   这里的kerner_name就是你创建的conda环境名称，也是你的ipython kernel name
export IPYKERNEL=chatglm3-demo
```
为什么要保持同步？因为在/demo_ci.py 中调用jupyter的桓环境是使用的IPYKERNEL环境变量，代码如下：
```
IPYKERNEL = os.environ.get('IPYKERNEL', 'chatglm3-demo')
```

所以当你创建的conda环境名称不一样时，需要设置
```
ipython kernel install --name <your_conda_env_name> --user
export IPYKERNEL=<your_conda_env_name>
```
这样才能确保/demo_ci.py 中能够调用到IPYKERNEL绘制图像 @a1120235890 @zRzRzRzRzRzRzR
